# ***Introduction: Business Problem***
The goal is to minimize travel for all Los Angeles residents in need of access to a local gym while promoting health across the city. The objective of this project is to determine optimal locations for a new gym or fitness center within Los Angeles. Specifically, the end result is to compile a list of Los Angeles neighborhoods that have no gyms or fitness centers within 3000 meters from their center points.

# **Data**

The following data resources will be used for this project:

The official list of neighborhoods in Los Angeles.

*   Description: Names of each Los Angeles neighborhood are listed. The scope of this project does not extend to the greater Los Angeles area and is confined to the official neighborhood boundaries under Los Angeles.
*   Data Source: geohub.lacity.org

Coordinates for each Los Angeles neighborhood.

*   Description: Latitude and longitude coordinates for the center of each Los Angeles neighborhood. These coordinates will be used to plot circle markers over a map of Los Angeles and to explore each of them.
* Source: geopy.geocoder package

Venue Data in proximity of each Los Angeles neighborhood.

*   Description: Using Foursquare API a search query will be done against the coordinates for each Los Angeles neighborhood and venue data that is within a radius of 3000 meters of each neighborhood will be generated. This venue data will then be filtered to account for only data related to gyms and fitness centers.
*   Data source: Foursquare API





In [3]:
# METHODOLOGY
%%capture
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files

!pip install geopy
!pip install geocoder
import geocoder
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 

from bs4 import BeautifulSoup # library to parse HTML and XML documents

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

!pip install folium
import folium # map rendering library

print('Libraries imported.')

# Data Gathering
Getting the link to the LA neighborhood data uploaded in a github repository.

In [4]:
url = "https://github.com/SakshamSomani/ssomani3/blob/main/Data_analysis_project/Strategic-New-Fitness-Centers-LA/LA_Neighborhood.csv" 

Passing the URL to a Pandas dataframe.

In [5]:
df_Neighbs = pd.read_html(url)

Pass the table of interest from the url into a Pandas data frame.

In [6]:
df_LA = df_Neighbs[0]

Dropping unnecessary columns.

In [7]:
df_LA.drop(["Unnamed: 0", "OBJECTID"], inplace=True, axis=1) 

Renaming columns

In [8]:
df_LA.rename(columns = {'name':'Neighborhood'}, inplace = True)

In [9]:

df_LA.head()

Neighborhood
0         Adams-Normandie
1                  Arleta
2       Arlington Heights
3         Atwater Village
4  Baldwin Hills/Crenshaw

In [10]:
df_LA.shape

(114, 1)

Geocoding LA Neighborhoods
Defining a function to retrieve each LA neighborhood's coordinates

In [11]:
def get_latlng(LAneighborhood):
    
    # initialize a variable to None
    lat_lng_coords = None
    
    # loop until all neighborhood coordinates are retrieved using geocoder package
    while(lat_lng_coords is None):
        g = geocoder.arcgis('{}, Los Angeles, California'.format(LAneighborhood))
        lat_lng_coords = g.latlng
    return lat_lng_coords

Calling the defined function to get the neighborhood coordinates and then storing the values into a list.

In [12]:
coordList = [ get_latlng(LAneighborhood) for LAneighborhood in df_LA["Neighborhood"].tolist() ]

Creating a temporary dataframe to populate the coordinates.

In [13]:
df_temp = pd.DataFrame(coordList, columns=['Latitude', 'Longitude'])

Merging the temporary dataframe columns with the df_LA.

In [14]:
df_LA['Latitude'] = df_temp['Latitude']
df_LA['Longitude'] = df_temp['Longitude']

In [15]:
df_LA.head()

Neighborhood   Latitude   Longitude
0         Adams-Normandie  33.901238 -118.299083
1                  Arleta  34.249050 -118.433490
2       Arlington Heights  34.039890 -118.325160
3         Atwater Village  34.119700 -118.258870
4  Baldwin Hills/Crenshaw  34.010428 -118.336776

Data Visualization: Mapping Los Angeles
Getting the coordinates of Los Angeles.

In [16]:
address = 'Los Angeles, CA'

geolocator = Nominatim(user_agent="ca_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Los Angeles are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Los Angeles are 34.0536909, -118.242766.


The geograpical coordinate of Los Angeles are 34.0536909, -118.242766.
Creating a map of Los Angeles by passing the latitude and longitude values for each neighborhood.

In [17]:
map_LA = folium.Map(location=[latitude, longitude], zoom_start=11)

# adding markers to map
for lat, lng, neighborhood in zip(df_LA['Latitude'], df_LA['Longitude'], df_LA['Neighborhood']):
    label = '{}'.format(neighborhood)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7).add_to(map_LA)  
    
map_LA

In [18]:
# saving the map as HTML file
map_LA.save('map_LA3.html')

# Exploring Neighborhood Venues With Foursquare API
Defining foursquare credentials and its version in order to access its database.

In [19]:
CLIENT_ID = 'H0ZIZZAEGBFDEGUJYC0XZKX4Z2ROHFX53WN0V5SENBVMDBVU' # my Foursquare ID
CLIENT_SECRET = 'D1HOVQ0X0UQV1JHXSZ0TIC0HFODDSELZH2WTWLJ0U5GQ5E1A' # my Foursquare Secret
VERSION = '20201105' # Foursquare API version
LIMIT = 100 # A default Foursquare API limit value

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: H0ZIZZAEGBFDEGUJYC0XZKX4Z2ROHFX53WN0V5SENBVMDBVU
CLIENT_SECRET:D1HOVQ0X0UQV1JHXSZ0TIC0HFODDSELZH2WTWLJ0U5GQ5E1A


Your credentails:
CLIENT_ID: H0ZIZZAEGBFDEGUJYC0XZKX4Z2ROHFX53WN0V5SENBVMDBVU
CLIENT_SECRET:D1HOVQ0X0UQV1JHXSZ0TIC0HFODDSELZH2WTWLJ0U5GQ5E1A
Defining a function to get top venues of each neighborhood within a 500 meter radius.

In [23]:
def getNearbyVenues(names, latitudes, longitudes, radius=2000):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # creating the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
        
        # making the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # returning only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

Running the above function to query each neighborhood through the foursquare API while passing the venue data to a new dataframe called losangeles_venues.



In [24]:
%%capture
losangeles_venues = getNearbyVenues(names=df_LA['Neighborhood'],
                                   latitudes=df_LA['Latitude'],
                                   longitudes=df_LA['Longitude']
                                  )


In [25]:
print(losangeles_venues.shape)
losangeles_venues.head()

(9635, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0  Adams-Normandie              33.901238             -118.299083   
1  Adams-Normandie              33.901238             -118.299083   
2  Adams-Normandie              33.901238             -118.299083   
3  Adams-Normandie              33.901238             -118.299083   
4  Adams-Normandie              33.901238             -118.299083   

                       Venue  Venue Latitude  Venue Longitude  \
0  Popeyes Louisiana Kitchen       33.901664      -118.295854   
1                  Starbucks       33.901643      -118.300039   
2                  Pho Daily       33.893948      -118.308582   
3                Spoon House       33.889974      -118.304900   
4                 Waba Grill       33.902347      -118.309726   

          Venue Category  
0    Fried Chicken Joint  
1            Coffee Shop  
2  Vietnamese Restaurant  
3     Italian Restaurant  
4    Japanese Restaurant

Checking how many venues were returned for each neighborhood.

In [26]:
losangeles_venues.groupby('Neighborhood').count()

Neighborhood Latitude  Neighborhood Longitude  Venue  \
Neighborhood                                                                   
Adams-Normandie                           100                     100    100   
Arleta                                     58                      58     58   
Arlington Heights                         100                     100    100   
Atwater Village                           100                     100    100   
Baldwin Hills/Crenshaw                     76                      76     76   
Bel-Air                                    62                      62     62   
Beverly Crest                              44                      44     44   
Beverly Grove                              24                      24     24   
Beverlywood                               100                     100    100   
Boyle Heights                             100                     100    100   
Brentwood                                  99                      99     99   
Broadway-Manchester                        19                      19     19   
Canoga Park                               100                     100    100   
Carthay                                   100                     100    100   
Central-Alameda                            52                      52     52   
Century City                              100                     100    100   
Chatsworth                                 97                      97     97   
Chatsworth Reservoir                       20                      20     20   
Chesterfield Square                        85                      85     85   
Cheviot Hills                             100                     100    100   
Chinatown                                 100                     100    100   
Cypress Park                               60                      60     60   
Del Rey                                   100                     100    100   
Downtown                                  100                     100    100   
Eagle Rock                                100                     100    100   
East Hollywood                            100                     100    100   
Echo Park                                 100                     100    100   
El Sereno                                  64                      64     64   
Elysian Park                              100                     100    100   
Elysian Valley                             56                      56     56   
Encino                                    100                     100    100   
Exposition Park                           100                     100    100   
Fairfax                                   100                     100    100   
Florence                                   95                      95     95   
Glassell Park                              81                      81     81   
Gramercy Park                              96                      96     96   
Granada Hills                              91                      91     91   
Green Meadows                             100                     100    100   
Griffith Park                              82                      82     82   
Hancock Park                              100                     100    100   
Hansen Dam                                 21                      21     21   
Harbor City                                96                      96     96   
Harbor Gateway                             91                      91     91   
Harvard Heights                           100                     100    100   
Harvard Park                              100                     100    100   
Highland Park                             100                     100    100   
Historic South-Central                     94                      94     94   
Hollywood                                 100                     100    100   
Hollywood Hills         

Checking how many unique categories can be curated from all the returned venues.

In [27]:
print('There are {} uniques categories.'.format(len(losangeles_venues['Venue Category'].unique())))

There are 411 uniques categories.


Creating a one hot encoding technique that turns each unique venue category into a column and resets the index to be by neighborhood.

In [28]:
# one hot encoding
losangeles_onehot = pd.get_dummies(losangeles_venues[['Venue Category']], prefix="", prefix_sep="")

# adding neighborhood column back to dataframe
losangeles_onehot['Neighborhood'] = losangeles_venues['Neighborhood'] 

# defining a list of column names
cols = losangeles_onehot.columns.tolist()
cols

# moving neighborhood column to the first column
cols.insert(0, cols.pop(cols.index('Neighborhood')))
cols

#applying .reindex() function to reorder
losangeles_onehot = losangeles_onehot.reindex(columns= cols)

#checking result
losangeles_onehot.head()

Neighborhood  ATM  Accessories Store  Adult Boutique  Airport  \
0  Adams-Normandie    0                  0               0        0   
1  Adams-Normandie    0                  0               0        0   
2  Adams-Normandie    0                  0               0        0   
3  Adams-Normandie    0                  0               0        0   
4  Adams-Normandie    0                  0               0        0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0               0                0                 0                    0   
1               0                0                 0                    0   
2               0                0                 0                    0   
3               0                0                 0                    0   
4               0                0                 0                    0   

   Amphitheater  Andhra Restaurant  Antique Shop  Aquarium  Arcade  \
0             0                  0             0         0       0   
1             0                  0             0         0       0   
2             0                  0             0         0       0   
3             0                  0             0         0       0   
4             0                  0             0         0       0   

   Argentinian Restaurant  Art Gallery  Art Museum  Arts & Crafts Store  \
0                       0            0           0                    0   
1                       0            0           0                    0   
2                       0            0           0                    0   
3                       0            0           0                    0   
4                       0            0           0                    0   

   Asian Restaurant  Astrologer  Athletics & Sports  Australian Restaurant  \
0                 0           0                   0                      0   
1                 0           0                   0                      0   
2                 0           0                   0                      0   
3                 0           0                   0                      0   
4                 0           0                   0                      0   

   Auto Dealership  Auto Garage  Automotive Shop  BBQ Joint  Baby Store  \
0                0            0                0          0           0   
1                0            0                0          0           0   
2                0            0                0          0           0   
3                0            0                0          0           0   
4                0            0                0          0           0   

   Bagel Shop  Bakery  Bank  Bar  Baseball Field  Baseball Stadium  \
0           0       0     0    0               0                 0   
1           0       0     0    0               0                 0   
2           0       0     0    0               0                 0   
3           0       0     0    0               0                 0   
4           0       0     0    0               0                 0   

   Basketball Court  Basketball Stadium  Beach  Bed & Breakfast  Beer Bar  \
0                 0                   0      0                0         0   
1                 0                   0      0                0         0   
2                 0                   0      0                0         0   
3                 0                   0      0                0         0   
4                 0                   0      0                0         0   

   Beer Garden  Beer Store  Big Box Store  Bike Rental / Bike Share  \
0            0           0              0                         0   
1            0           0              0                         0   
2            0           0              0                         0   
3            0           0              0                         0   
4            0           0              0                         0   

   Bike Trail

In [29]:
#viewing the new dataframe size
losangeles_onehot.shape

(9635, 411)

Grouping the rows by neighborhood and taking the mean for the frequency of occurrence for each venue category.

In [31]:
losangeles_grouped = losangeles_onehot.groupby('Neighborhood').mean().reset_index()
losangeles_grouped.head()

Neighborhood   ATM  Accessories Store  Adult Boutique  Airport  \
0         Adams-Normandie  0.00                0.0             0.0      0.0   
1                  Arleta  0.00                0.0             0.0      0.0   
2       Arlington Heights  0.00                0.0             0.0      0.0   
3         Atwater Village  0.01                0.0             0.0      0.0   
4  Baldwin Hills/Crenshaw  0.00                0.0             0.0      0.0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0             0.0              0.0               0.0             0.000000   
1             0.0              0.0               0.0             0.000000   
2             0.0              0.0               0.0             0.010000   
3             0.0              0.0               0.0             0.030000   
4             0.0              0.0               0.0             0.013158   

   Amphitheater  Andhra Restaurant  Antique Shop  Aquarium  Arcade  \
0           0.0                0.0           0.0       0.0     0.0   
1           0.0                0.0           0.0       0.0     0.0   
2           0.0                0.0           0.0       0.0     0.0   
3           0.0                0.0           0.0       0.0     0.0   
4           0.0                0.0           0.0       0.0     0.0   

   Argentinian Restaurant  Art Gallery  Art Museum  Arts & Crafts Store  \
0                     0.0         0.00         0.0                  0.0   
1                     0.0         0.00         0.0                  0.0   
2                     0.0         0.01         0.0                  0.0   
3                     0.0         0.00         0.0                  0.0   
4                     0.0         0.00         0.0                  0.0   

   Asian Restaurant  Astrologer  Athletics & Sports  Australian Restaurant  \
0              0.04         0.0                 0.0                    0.0   
1              0.00         0.0                 0.0                    0.0   
2              0.00         0.0                 0.0                    0.0   
3              0.01         0.0                 0.0                    0.0   
4              0.00         0.0                 0.0                    0.0   

   Auto Dealership  Auto Garage  Automotive Shop  BBQ Joint  Baby Store  \
0              0.0     0.000000              0.0       0.03         0.0   
1              0.0     0.000000              0.0       0.00         0.0   
2              0.0     0.000000              0.0       0.02         0.0   
3              0.0     0.000000              0.0       0.00         0.0   
4              0.0     0.013158              0.0       0.00         0.0   

   Bagel Shop    Bakery      Bank   Bar  Baseball Field  Baseball Stadium  \
0         0.0  0.020000  0.020000  0.00             0.0               0.0   
1         0.0  0.000000  0.034483  0.00             0.0               0.0   
2         0.0  0.010000  0.010000  0.02             0.0               0.0   
3         0.0  0.030000  0.000000  0.01             0.0               0.0   
4         0.0  0.013158  0.026316  0.00             0.0               0.0   

   Basketball Court  Basketball Stadium  Beach  Bed & Breakfast  Beer Bar  \
0               0.0                 0.0    0.0              0.0       0.0   
1               0.0                 0.0    0.0              0.0       0.0   
2               0.0                 0.0    0.0              0.0       0.0   
3               0.0                 0.0    0.0              0.0       0.0   
4               0.0                 0.0    0.0              0.0       0.0   

   Beer Garden  Beer Store  Big Box Store  Bike Rental / Bike Share  \
0          0.0         0.0       0.000000                       0.0   
1          0.0         0.0       0.017241                       0.0   
2          0.0         0.0       0.000000                       0.0   
3          0.0         0.0       0.000000                       0.0   
4 

In [32]:
#confirming new size
losangeles_grouped.shape

(114, 411)

Clustering Neighborhoods by Gym & Fitness Center Data
Getting a count of neighborhoods with at least 1 occurance of a gym or fitness center near them.

In [33]:
len(losangeles_grouped[losangeles_grouped["Gym / Fitness Center"] > 0])

51

Creating a new data frame for the LA neighborhood data on Gyms and Fitness Centers.

In [34]:
LA_gyms = losangeles_grouped[["Neighborhood","Gym / Fitness Center"]]


In [35]:
LA_gyms.head()

Neighborhood  Gym / Fitness Center
0         Adams-Normandie                  0.00
1                  Arleta                  0.00
2       Arlington Heights                  0.02
3         Atwater Village                  0.02
4  Baldwin Hills/Crenshaw                  0.00

Getting the K-Means Clustering technique setup.

In [36]:
# setting number of clusters
kclusters = 5

losangeles_grouped_clustering = LA_gyms.drop('Neighborhood', 1)

# running k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(losangeles_grouped_clustering)

# checking cluster labels generated for each row in the dataframe
kmeans.labels_[0:100] 

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only
  after removing the cwd from sys.path.


array([2, 2, 4, 4, 2, 2, 2, 2, 2, 2, 0, 2, 2, 1, 2, 2, 4, 0, 2, 1, 2, 2,
       1, 4, 1, 2, 4, 2, 2, 2, 1, 2, 1, 1, 3, 1, 2, 2, 2, 2, 2, 1, 2, 1,
       4, 2, 2, 2, 2, 2, 2, 1, 2, 2, 2, 2, 2, 2, 2, 4, 1, 3, 4, 1, 2, 2,
       2, 4, 2, 2, 2, 3, 2, 1, 2, 1, 1, 2, 1, 2, 1, 4, 2, 2, 3, 4, 2, 0,
       4, 2, 2, 4, 2, 2, 2, 1, 1, 1, 0, 2], dtype=int32)

In [37]:
LA_merged = LA_gyms.copy()
# adding cluster labels column 
LA_merged["Cluster Labels"] = kmeans.labels_

LA_merged.head() # check the last column

Neighborhood  Gym / Fitness Center  Cluster Labels
0         Adams-Normandie                  0.00               2
1                  Arleta                  0.00               2
2       Arlington Heights                  0.02               4
3         Atwater Village                  0.02               4
4  Baldwin Hills/Crenshaw                  0.00               2

Merging df_LA with LA_merged to add latitude/longitude for each neighborhood.

In [38]:
LA_merged = LA_merged.join(df_LA.set_index("Neighborhood"), on="Neighborhood")

print(LA_merged.shape)
LA_merged.head() # check the last columns

(114, 5)


Neighborhood  Gym / Fitness Center  Cluster Labels   Latitude  \
0         Adams-Normandie                  0.00               2  33.901238   
1                  Arleta                  0.00               2  34.249050   
2       Arlington Heights                  0.02               4  34.039890   
3         Atwater Village                  0.02               4  34.119700   
4  Baldwin Hills/Crenshaw                  0.00               2  34.010428   

    Longitude  
0 -118.299083  
1 -118.433490  
2 -118.325160  
3 -118.258870  
4 -118.336776

Visualizing the resulting clusters.

In [39]:
# create map
map_clusters = folium.Map(location=[latitude, longitude], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(LA_merged['Latitude'], LA_merged['Longitude'], LA_merged['Neighborhood'], LA_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[int(cluster)-1],
        fill=True,
        fill_color=rainbow[int(cluster)-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

In [40]:
map_clusters.save('map_clusters.html')

# **Results**

## **Cluster 0** 
(Red)

In [41]:
cluster0 = LA_merged.loc[LA_merged['Cluster Labels'] == 0]
cluster0

Neighborhood  Gym / Fitness Center  Cluster Labels  Latitude  \
10             Brentwood              0.040404               0  34.06626   
17  Chatsworth Reservoir              0.050000               0  34.22750   
87           Studio City              0.040000               0  34.14453   
98                Venice              0.040000               0  33.98754   

    Longitude  
10 -118.47033  
17 -118.63148  
87 -118.39561  
98 -118.47215

In [42]:
cluster0.shape

(4, 5)

Cluster 0 groups together 28 neighborhoods with a low concentration of fitness centers in proximity. The venue data for these neighborhoods generated at least 1 occurrence of a fitness center within a 2000 meter radius from their center points.

## **Cluster 1** 
(Purple)

In [43]:
cluster1 = LA_merged.loc[LA_merged['Cluster Labels'] == 1]
cluster1

Neighborhood  Gym / Fitness Center  Cluster Labels   Latitude  \
13            Carthay              0.010000               1  34.056120   
19      Cheviot Hills              0.010000               1  34.034030   
22            Del Rey              0.010000               1  33.990820   
24         Eagle Rock              0.010000               1  34.139270   
30             Encino              0.010000               1  34.165380   
32            Fairfax              0.010000               1  34.076100   
33           Florence              0.010526               1  33.974750   
35      Gramercy Park              0.010417               1  34.034053   
41        Harbor City              0.010417               1  33.797810   
43    Harvard Heights              0.010000               1  34.043390   
51     Jefferson Park              0.011628               1  34.032640   
60          Mar Vista              0.010000               1  34.004680   
63      Mission Hills              0.014286               1  34.273970   
73     Pico-Robertson              0.010000               1  34.055750   
75        Playa Vista              0.010000               1  33.973950   
76      Playa del Rey              0.014286               1  33.959250   
78        Rancho Park              0.010000               1  34.045070   
80          San Pedro              0.011494               1  33.737880   
95        Valley Glen              0.010000               1  34.189910   
96     Valley Village              0.012500               1  34.164880   
97           Van Nuys              0.010000               1  34.184390   
104        West Adams              0.010000               1  34.028510   
105        West Hills              0.010000               1  34.200360   
106  West Los Angeles              0.010000               1  34.046470   
107       Westchester              0.010000               1  33.959730   
108          Westlake              0.010000               1  34.053710   
109          Westwood              0.010000               1  34.062080   
111    Windsor Square              0.010000               1  34.068960   

      Longitude  
13  -118.373410  
19  -118.410560  
22  -118.420620  
24  -118.210870  
30  -118.527110  
32  -118.361220  
33  -118.245390  
35  -118.312335  
41  -118.302480  
43  -118.309030  
51  -118.318020  
60  -118.429130  
63  -118.467830  
73  -118.380000  
75  -118.430380  
76  -118.447990  
78  -118.419240  
80  -118.283730  
95  -118.449530  
96  -118.398030  
97  -118.446520  
104 -118.352790  
105 -118.629330  
106 -118.433950  
107 -118.413690  
108 -118.271300  
109 -118.445450  
111 -118.318710

In [44]:
cluster1.shape

(28, 5)

Cluster 1 groups together 68 neighborhoods with no existence of fitness centers in proximity. The venue data for these neighborhoods generated 0 occurrences of a fitness center within a 2000 meter radius from their center points.

## **Cluster 2**
(Blue)

In [45]:
cluster2 = LA_merged.loc[LA_merged['Cluster Labels'] == 2]
cluster2

Neighborhood  Gym / Fitness Center  Cluster Labels   Latitude  \
0           Adams-Normandie                   0.0               2  33.901238   
1                    Arleta                   0.0               2  34.249050   
4    Baldwin Hills/Crenshaw                   0.0               2  34.010428   
5                   Bel-Air                   0.0               2  34.083800   
6             Beverly Crest                   0.0               2  34.096533   
7             Beverly Grove                   0.0               2  34.095448   
8               Beverlywood                   0.0               2  34.051380   
9             Boyle Heights                   0.0               2  34.033900   
11      Broadway-Manchester                   0.0               2  34.398960   
12              Canoga Park                   0.0               2  34.202390   
14          Central-Alameda                   0.0               2  33.997420   
15             Century City                   0.0               2  34.059440   
18      Chesterfield Square                   0.0               2  33.986682   
20                Chinatown                   0.0               2  34.062290   
21             Cypress Park                   0.0               2  34.094480   
25           East Hollywood                   0.0               2  34.090860   
27                El Sereno                   0.0               2  34.076850   
28             Elysian Park                   0.0               2  34.082500   
29           Elysian Valley                   0.0               2  34.071728   
31          Exposition Park                   0.0               2  34.012220   
36            Granada Hills                   0.0               2  34.264420   
37            Green Meadows                   0.0               2  33.812340   
38            Griffith Park                   0.0               2  34.123270   
39             Hancock Park                   0.0               2  34.068920   
40               Hansen Dam                   0.0               2  34.260560   
42           Harbor Gateway                   0.0               2  33.854126   
45            Highland Park                   0.0               2  34.109820   
46   Historic South-Central                   0.0               2  34.019790   
47                Hollywood                   0.0               2  34.101560   
48          Hollywood Hills                   0.0               2  34.105240   
49     Hollywood Hills West                   0.0               2  34.120060   
50                Hyde Park                   0.0               2  33.989040   
52                Koreatown                   0.0               2  34.057790   
53              Lake Balboa                   0.0               2  34.207160   
54        Lake View Terrace                   0.0               2  34.274530   
55                Larchmont                   0.0               2  34.081330   
56             Leimert Park                   0.0               2  34.008380   
57          Lincoln Heights                   0.0               2  34.070030   
58                Los Feliz                   0.0               2  34.105410   
64        Montecito Heights                   0.0               2  34.091980   
65         Mount Washington                   0.0               2  34.099040   
66              North Hills                   0.0               2  34.235630   
68               Northridge                   0.0               2  34.235120   
69        Pacific Palisades                   0.0               2  34.047100   
70                  Pacoima                   0.0               2  34.261080   
72            Panorama City                   0.0               2  34.222830   
74               Pico-Union                   0.0               2  34.040050   
77             Porter Ranch                   0.0               2  34.280270   
79                   Reseda                   0.0               2  34.193840   
82          Sep

In [46]:
cluster2.shape

(63, 5)

Cluster 2 groups together 4 neighborhoods with a mid-high concentration of fitness centers in proximity. The venue data for these neighborhoods generated 3 occurrences of a fitness center within a 2000 meter radius from their center points.

## **Cluster 3**
(Green)

In [47]:
cluster3 = LA_merged.loc[LA_merged['Cluster Labels'] == 3]
cluster3

Neighborhood  Gym / Fitness Center  Cluster Labels  Latitude  Longitude
34    Glassell Park              0.024691               3  34.11925 -118.23039
61         Mid-City              0.030000               3  34.04021 -118.34761
71            Palms              0.030303               3  34.01988 -118.42093
84     Sherman Oaks              0.030000               3  34.15106 -118.44434
113  Woodland Hills              0.030000               3  34.16889 -118.61148

In [48]:
cluster3.shape

(5, 5)

Cluster 3 groups together 9 neighborhoods with a low-mid concentration of fitness centers in proximity. The venue data for these neighborhoods generated at least 2 occurrences of a fitness center within a 2000 meter radius from their center points.

## **Cluster 4**
(Orange)

In [49]:
cluster4 = LA_merged.loc[LA_merged['Cluster Labels'] == 4]
cluster4

Neighborhood  Gym / Fitness Center  Cluster Labels   Latitude  \
2    Arlington Heights              0.020000               4  34.039890   
3      Atwater Village              0.020000               4  34.119700   
16          Chatsworth              0.020619               4  34.257250   
23            Downtown              0.020000               4  34.048340   
26           Echo Park              0.020000               4  34.078080   
44        Harvard Park              0.020000               4  34.144400   
59   Manchester Square              0.019231               4  33.960010   
62        Mid-Wilshire              0.020000               4  34.040051   
67     North Hollywood              0.020000               4  34.169820   
81            Sawtelle              0.020000               4  34.034610   
85         Silver Lake              0.020000               4  34.086680   
88          Sun Valley              0.016949               4  34.215600   
91             Tarzana              0.020000               4  34.175290   
110         Wilmington              0.016667               4  33.789890   

     Longitude  
2   -118.32516  
3   -118.25887  
16  -118.59101  
23  -118.25564  
26  -118.26066  
44  -118.25297  
59  -118.29992  
62  -118.32582  
67  -118.37899  
81  -118.44817  
85  -118.27023  
88  -118.37035  
91  -118.55010  
110 -118.26323

In [50]:
cluster4.shape

(14, 5)

Cluster 4 groups together 2 neighborhoods with a high concentration of fitness centers in proximity. The venue data for these neighborhoods generated at least 4 occurrences of a fitness center within a 2000 meter radius from their center points.

# **Discussion**
Based on the clustering data over the map of Los Angeles, the majority of neighborhoods under Los Angeles county lack access to nearby fitness centers. Very few neighborhoods harbor a mid to high concentration of local fitness centers present. In fact, clusters 2, 3 and 4, which contain a mid to high concentration of fitness centers, can be combined and they would only total 17 neighborhoods out of the 114. Unsurprisingly, clusters 2, 3, and 4 contain suburban neighborhoods with a higher income population. Judging also by their higher concentration of fitness centers, it is safe to recommend to avoid building new fitness centers in these areas since they already have competition established. Even more so, the intent is to give more local access to neighborhoods that lack fitness centers.

Continuing on, the second largest cluster is cluster 0, which list neighborhoods with at least one occurrence of a fitness center in proximity. Cluster 0 contained 30 neighborhoods out of the 114 in Los Angeles and geographically was concentrated towards Western Los Angeles and North West Los Angeles. Competition for memberships is probably not high in these neighborhoods so they can be considered as a secondary tier of neighborhoods for building more fitness centers around.

Every neighborhood that did not generate an occurence of a fitness center within a 2000 meter radius from their center point was assigned to Cluster 1. Cluster 1 had a total of 67 out of 114 neighborhoods assigned to its cluster. Geographically, the neighborhoods under cluster 1 that are located in the most southern side of Los Angeles have the least availability to access a fitness center in proximity.This discrepancy could be due to the fact high concentration of low income residents reside in the southern side of Los Angeles which in the past might have deterred property developers from establishing fitness centers there.

In Addition, there are also several other notable geographic concentrations from cluster 1 that display a lack of available neighborhoods such as East Los Angeles, Central Los Angeles, and the most northern regions of Los Angeles that hug San Fernando. These city regions from cluster 1 are also high population dense areas so there is definitely a high probability of demand for more convenient fitness centers in these areas.


# **Conclusion**
Despite there being some limitations as far as how much venue data can be queried from Foursquare, enough venue data was gathered where an accurate mean frequency of occurrence for gyms and fitness centers can be obtained for each Los Angeles neighborhood. From these mean frequency of occurrences for gyms and fitness centers, five unique clusters were able to be made and each held notable geographic insights. In Particular the most stand out cluster that can be referred to as the ideal list of neighborhoods to target for building new fitness centers is cluster 1. Cluster 1 revealed that there are at least three big and populous regions in Los Angeles that have zero fitness centers established and they are Southern Los Angeles, East los Angeles, and Central Los Angeles. While these regions have historically housed low-income residents, this factor should not limit property developers from pursuing to build fitness centers in these regions. In fact, property developers should aim to build a market of low-cost membership fitness centers in these regions. Better yet, the city of Los Angeles can serve the health needs of its low-income residents better by building community fitness recreational centers in these neighborhoods as well. Overall, there is a market and need to build more fitness centers for the neighborhoods listed under cluster 1.